DatadogCore/Sources/Core/Storage/Files/Directory.swift

```diff
@@ -106,9 +110,7 @@ internal struct Directory: DirectoryProtocol {
     /// Creates file with given name.
     func createFile(named fileName: String) throws -> File {
         let fileURL = url.appendingPathComponent(fileName, isDirectory: false)
-        guard FileManager.default.createFile(atPath: fileURL.path, contents: nil, attributes: nil) == true else {
-            throw InternalError(description: "Cannot create file at path: \(fileURL.path)")
-        }
+        try Data.empty.write(to: fileURL, options: .atomic)
```

Comment:
question/ Do we have any insights into what kind of errors it can throw? What can we expect to see in telemetry?

Reply:
Assuming this is 1:1 mapping of https://developer.apple.com/documentation/foundation/nsdata/1414800-writetofile and it uses syscall to perform file IO, I can expect https://man7.org/linux/man-pages/man2/write.2.html#top_of_page:~:text=are%20not%20specified.-,ERRORS,-top errors.